django/db/models/expressions.py

```diff
@@ -366,7 +366,12 @@ def __eq__(self, other):
 
     def __hash__(self):
         path, args, kwargs = self.deconstruct()
-        return hash((path,) + args + tuple(kwargs.items()))
+        try:
+            return hash((path,) + args + tuple(kwargs.items()))
+        except TypeError:
+            # args or kwargs may include a mutable object, in this case we fall back to
```

Comment:
This doesn't look safe... now two different expressions could hash to the same value?

Reply:
The `__hash__` promise is `a == b implies hash(a) == hash(b)`. The implication going the opposite way is impossible.

As far as I can remember, the patch works without this change, due to get_db_prep_value being called before `__hash__`.